ds3_java_cli/src/test/java/com/spectralogic/ds3cli/Ds3Cli_Test.java

```diff
@@ -3009,5 +3006,19 @@ public void getDetailedObjectsBadArgs() throws Exception {
         command.init(args);
     }
 
+    @Test
+    public void parseRelativeDate() throws Exception {
+        final String input = "d1.h2.m3.s4";
+        final long diff = Utils.dateDiffToSeconds(input);
+        assertEquals(diff, (24 * 60 * 60) + (2 * 60 * 60) + (3 * 60) + 4);
+    }
+
+    @Test
+    public void parseAbsoluteDate() throws Exception {
+        final String input = "Y1960.M5.D26";
+        final long parse = Utils.parseParamDate(input).getTime();
+        final long construct =  Constants.DATE_FORMAT.parse("1960-05-26T00:00:00.000Z").getTime();
+        assertEquals(parse, construct);
+    }
 
```

Comment:
I had a note on my other branch about this. The Java DateFormat parser is oddly tolerant. It parses 2016-99-88T00:00:00.000Z as Sun May 26 18:00:00 MDT 2024.

I do now throw an Exception on an unrecognized token and have a unit test.